static/js/settings_account.js

```diff
@@ -562,8 +562,7 @@ export function set_up() {
             channel.del({
                 url: "/json/users/me",
                 success() {
-                    dialog_widget.hide_dialog_spinner();
-                    overlays.close_modal("#dialog_widget_modal");
+                    dialog_widget.close_modal();
                     window.location.href = "/login/";
```

Comment:
This commit makes too many scattered changes like this which need separate review.  You should pull the extraction of a `dialog_widget.close_modal()` function and migrating to it into preparatory commit series (or even a preparatory mini PR) to help manage the complexity of this migration.